### PR TITLE
Improves supplypack code to support repeats

### PR DESCRIPTION
**Pull request by ExcessiveUseOfCobblestone**
## About The Pull Request

allows supplypacks to use the power of assoc lists to spawn multiple of the same item (vs previous functionality of writing it multiple times repeatedly)

WARNING: I initially did this on an old branch and then tried to update it so may be some errors associated. Converting things to 1 line was done manually so v likely some mistypes. I also havent coded in a hot sec so 😈 

Tested by spawning a cargo pack and seeing if it gives same result as old code. Did this with a random normal crate and inducer crate (this crate has parameters included).

## Why It's Good For The Game

reading is hard and writing the same thing multiple times is error-prone and annoying

## Changelog

not player facing

### CODE REVIEW

**Comment by Mothblocks** [comment]
Above and beyond: Unit test to ensure every crate only has one of each typepath (to ensure all new crates use this system)

**Review by Mothblocks — APPROVED** [review]

**Comment by ExcessiveUseOfCobblestone** [comment]
important to note my gbp will be broken upon merging this

**Comment by Mothblocks** [comment]
waht

**Comment by ExcessiveUseOfCobblestone** [comment]
look at my balance

**Comment by Mothblocks** [comment]
:(

File: code/modules/cargo/packs.dm [hunk 1]
```diff
@@ -153,18 +140,12 @@
...
-					/obj/item/extinguisher/advanced,
-					/obj/item/extinguisher/advanced,
-					/obj/item/clothing/head/hardhat/red,
-					/obj/item/clothing/head/hardhat/red)
+	contains = list(/obj/item/clothing/suit/utility/fire/firefighter = 2,
+					/obj/item/clothing/mask/gas = 2,
+					/obj/item/flashlight = 2,
+					/obj/item/tank/internals/oxygen/red = 2,
+					/obj/item/extinguisher/advanced = 2,
+					/obj/item/clothing/head/hardhat/red = 2)
```
[the remark this answers is not in the document]
Comment (ExcessiveUseOfCobblestone) [reply code/modules/cargo/packs.dm:148]: miss git isnt seeing a difference when i convert it unfortunately unless im misreading

[the remark this answers is not in the document]
Comment (Mothblocks) [reply code/modules/cargo/packs.dm:148]: as in

```dm
contains = list(
    item1, // go up only one indent
    item2, // trailing comma
) // parenthese at the end, back at the same indent level as list
```

[the remark this answers is not in the document]
Comment (ExcessiveUseOfCobblestone) [reply code/modules/cargo/packs.dm:148]: mental

**Pull request merged**